clusterman/signals/pending_pods_signal.py

```diff
@@ -65,3 +76,20 @@ def _get_resource_request(
             logger.info(f"Pending pods adding resource request: {resource_request} (multiplier {multiplier})")
 
         return resource_request + allocated_resources
+
+    def _get_resource_request_v2(
+        self,
+        allocated_resources: ClustermanResources,
+        total_resources: ClustermanResources,
+        target_capacity_margin: float,
+        pending_pods: Optional[List[KubernetesPod]] = None,
+    ) -> SignalResourceRequest:
+        """Given a list of metrics, construct a resource request based on the most recent
+        data for allocated and pending pods"""
+
+        pending_pods = pending_pods or []
+
+        if len(pending_pods) > 0:
+            return total_resources * (1 + target_capacity_margin)
```

Comment:
we need better calculation here. pending pods may request more than margin

Reply:
What are we trying to achieve here? We want clusterman to add new hosts whenever there's pending pods right?

How about if we added a fixed amount of resource request, enough to scale up by at least N nodes in the pool. This would prevent overscaling like we had yesterday but also guarantee that we scale up _by something_ whenever there's pending pods?

Reply:
(Bear in mind this is quite possibly a terrible idea, I've not had coffee yet)